internal/command/machine/restart.go

```diff
@@ -70,11 +70,24 @@ func runMachineRestart(ctx context.Context) error {
 		client  = client.FromContext(ctx).API()
 	)
 
-	app, err := client.GetAppCompact(ctx, appName)
+	app, err := appFromMachineOrName(ctx, args[0], appName)
 	if err != nil {
 		return fmt.Errorf("could not get app: %w", err)
 	}
 
+	// Ensure that all machines are on the same app
+	for _, machineID := range args {
+
+		machine, err := client.GetMachine(ctx, machineID)
```

Comment:
This is making a Web API call per machine where few lines below there is another loop that does the same but hitting Machines API endpoint:  https://github.com/superfly/flyctl/blob/1e084c967e42042a20b43f3c9d80ca900accd237/internal/command/machine/restart.go#L114-L124

It will return a 404 if the machine doesn't belong to the app tho, which should be enough, isn't it?

I would check [line 144](https://github.com/superfly/flyctl/blob/1e084c967e42042a20b43f3c9d80ca900accd237/internal/command/machine/restart.go#L114) too, that is pulling the flaps client (which needs an app to operate) that somehow could be different (or always the same) than what we get in https://github.com/superfly/flyctl/blob/1e084c967e42042a20b43f3c9d80ca900accd237/internal/command/machine/restart.go#L73

Reply:
I double-hit the API because the second one only says something like "machine not found," which could confusingly imply that either the machine ID was wrong *or* that the machine wasn't in the app. When the app is specified implicitly by the first machine, this gets confusing.

That said, I guess it would be more reasonable to just catch the error in the second loop, and if it's a machine not found error, turn the error into something like "machine {id} not found. perhaps this machine is not in the app {app}"

Reply:
> I would check [line 144](https://github.com/superfly/flyctl/blob/1e084c967e42042a20b43f3c9d80ca900accd237/internal/command/machine/restart.go#L114) too, that is pulling the flaps client (which needs an app to operate) that somehow could be different (or always the same) than what we get in

I'm sorry, I'm not sure what you're getting at here. I rechecked the flow and I don't think I'm missing anything 
The app is derived from `appFromMachineOrName`. that's passed into `apps.BuildContext`, which initializes a bunch of things including the flaps client, and then `flaps.FromContext(ctx)` retrieves that flaps client.

At least with my understanding, this should all be fairly normal usage?

Reply:
> That said, I guess it would be more reasonable to just catch the error in the second loop, and if it's a machine not found error, turn the error into something like "machine {id} not found. perhaps this machine is not in the app {app}"

Yeap, good idea. That is a good tradeoff IMO.

> I rechecked the flow and I don't think I'm missing anything
The app is derived from appFromMachineOrName. that's passed into apps.BuildContext, which initializes a bunch of things including the flaps client, and then flaps.FromContext(ctx) retrieves that flaps client.

You just taught me how it works! thx